src/App.tsx

```diff
@@ -8,55 +8,163 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { customFetch } from './utils/fetchClient';
+import { User } from './types/User';
+import { ErrorType } from './types/ErrorType';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isErrorType, setErrorType] = useState<ErrorType | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isPostListVisible, setIsPostListVisible] = useState(false);
+  const [isLoaderVisible, setIsLoaderVisible] = useState(false);
+  const [isCommentFetch, setIsCommentFetch] = useState(false);
+  const [isCommentListHidden, setIsCommentListHidden] = useState(true);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isOpenFormForComment, setOpenFormForComment] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const loadingErrors = [ErrorType.USERS, ErrorType.POSTS];
+
+  const getUsers = async () => {
+    try {
```

Comment:
Consider implementing error handling for the case when the fetchUsers request fails silently (i.e., it doesn't throw an error but doesn't return the expected data). For instance, you could check if fetchUsers is an array before setting the state.